x/stake/keeper/validator.go

```diff
@@ -283,6 +290,13 @@ func (k Keeper) GetBondedValidatorsByPower(ctx sdk.Context) []types.Validator {
 	return validators[:i] // trim
 }
 
+// returns an iterator for the current validator power store
+func (k Keeper) ValidatorsPowerStoreIterator(ctx sdk.Context) (iterator sdk.Iterator) {
+	store := ctx.KVStore(k.storeKey)
+	iterator = sdk.KVStoreReversePrefixIterator(store, ValidatorsByPowerIndexKey)
+	return iterator
```

Comment:
Same this design pattern doesn't quite make sense to me

Reply:
It's needed to do the `PowerStoreInvariant` check.  It needs to return just an iterator, so that way the invariant function can go through the iterator itself (and have access to the `iterator.Key`s)